### PR TITLE
Remove github helper in favor of scmPlugin

### PR DESCRIPTION
**Pull request by petey**
- Now uses scmPlugin instead of built in github specific code
- Refactored tests calling `new Promise`
- Added support for adding build log link on create build

### CODE REVIEW

**Comment by d2lam** [comment]
Don't need this dependency anymore https://github.com/screwdriver-cd/models/blob/master/package.json#L45

File: lib/build.js [hunk 1]
```diff
@@ -192,22 +195,19 @@ class BuildModel extends BaseModel {
...
                 // update github
-                .then(admin =>
-                    updateSCM({
-                        user: admin,
-                        scmUrl: pipeline.scmUrl,
-                        sha: this.sha,
-                        state: 'failure'
-                    })
-                )
+                .then(() => this.updateCommitStatus(pipeline))
```
Comment (nkatzman) [inline lib/build.js]: When the launcher notifies the builds webhook, it should also be updating the commitStatus.

Is there a need to do it here as well?

Comment (petey) [inline lib/build.js]: I thought about that, and originally had it work that way, but I figured there wasn't a real good reason to do it twice, so took the update call out.

**Comment by nkatzman** [comment]
👍

**Comment by tkyi** [comment]
👍 
giphy-2

**Pull request merged**